### ISSUE TITLE
After upgrading to Remix 1.12.0: TypeError: Cannot read properties of undefined (reading 'remixRequestHandlerPath')

### ISSUE THREAD

**Issue by giopetris** [remix-run/remix#5428]
### What version of Remix are you using?

1.12.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

I'm not sure how to reproduce this, but this error started after I upgrade remix `1.11.0` to `1.12.0`, I tried to reproduce by creating a project from scratch using `npx create-remix@latest` but it worked fine. 

### Expected Behavior

Run the remix app.

### Actual Behavior

It is crashing with the error:

```
TypeError: Cannot read properties of undefined (reading 'remixRequestHandlerPath')
    at requestHandler (/dashboard/node_modules/@remix-run/server-runtime/dist/server.js:38:94)
    at /dashboard/node_modules/@remix-run/express/dist/server.js:39:28
    at /dashboard/node_modules/@remix-run/serve/dist/index.js:47:7
    at Layer.handle [as handle_request] (/dashboard/node_modules/express/lib/router/layer.js:95:5)
    at next (/dashboard/node_modules/express/lib/router/route.js:144:13)
    at next (/dashboard/node_modules/express/lib/router/route.js:140:7)
    at next (/dashboard/node_modules/express/lib/router/route.js:140:7)
    at next (/dashboard/node_modules/express/lib/router/route.js:140:7)
    at next (/dashboard/node_modules/express/lib/router/route.js:140:7)
    at next (/dashboard/node_modules/express/lib/router/route.js:140:7)
```

This is my `remix.config.js`:

```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  ignoredRouteFiles: ["**/.*"],
  future: {
    v2_routeConvention: true,
  },
};
```

------------

After digging a bit, I changed my `remix.config.js` to the following, the error stopped happening, but, got a new one 😅 : 

```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  ignoredRouteFiles: ["**/.*"],
  future: {
    v2_routeConvention: true,
    unstable_dev: { remixRequestHandlerPath: "" }, // <--- Added this line
  },
};
```

Now my browser shows `Upgrade Required`:
<img width="195" alt="CleanShot 2023-02-11 at 11 01 04@2x" src="https://user-images.githubusercontent.com/1332141/218262063-82a78099-0e2a-428a-a2da-81570aa38a82.png">

Nothing on the terminal.

**Comment by alexblack** [comment]
I'm hitting this too, trying to upgrade from 1.8.2 -> 1.13.0

```
[1] TypeError: Cannot read properties of undefined (reading 'remixRequestHandlerPath')
[1]     at requestHandler (/Users/alex/dev/micro/webapp/node_modules/@remix-run/server-runtime/dist/server.js:38:94)
[1]     at /Users/alex/dev/micro/webapp/node_modules/@remix-run/express/dist/server.js:39:28
[1]     at /Users/alex/dev/micro/webapp/node_modules/@sentry/src/utils/serverAdapters/express.ts:85:1
```

**Comment by alexblack** [comment]
Adding this line:

```
module.exports = {
  ignoredRouteFiles: ["**/.*"],
  future: {
    unstable_dev: { remixRequestHandlerPath: "" }, // <--- Added this line
  },
};
```

Didn't seem to help me, no change.

**Comment by alexblack** [comment]
ah its this line:

```
if (mode$1 === "development" && unstable_dev !== false && url.pathname === (unstable_dev.remixRequestHandlerPath ?? "") + "/__REMIX_ASSETS_MANIFEST") {
```

If I change it to:

```
if (mode$1 === "development" && unstable_dev !== false && url.pathname === (unstable_dev?.remixRequestHandlerPath ?? "") + "/__REMIX_ASSETS_MANIFEST") {
```

Then it works

**Comment by alexblack** [comment]
https://github.com/remix-run/remix/blob/main/packages/remix-server-runtime/server.ts#L60

**Comment by pcattori** [comment]
Culprit is here: https://github.com/remix-run/remix/blob/7c5b6b55bed60ca2d02f52083bf9226db5a9ff8d/packages/remix-server-runtime/build.ts#L16

Should have been:
```ts
  future?: FutureConfig;
```

Symptom of the duplicated types we have across packages.

**Comment by pcattori** [comment]
From @brophdawg11 on discord:

Ah thank you for this additional context @derekxw!  This lack of build errors was fixed in https://github.com/remix-run/remix/pull/5441 but I don't think it had been made apparent that this could be an underlying cause of these Cannot read properties of undefined (reading 'v2_errorBoundary')" errors.
For anyone else running into this issue in the future - please run remix build after upgrading all remix dependencies and see if it surfaces any build errors.  The merged fix to also ensure these errors show during remix dev should be released in 1.14.0

Also from @brophdawg11 on discord:

This usually comes down to one of 2 issues: (1) not all of your remix deps are on the same version or (2) one of your builds is failing and it's not getting logged properly by npm run dev.  There's more context in this thread: https://discord.com/channels/770287896669978684/1067867784161263686/1075162483028926568

**Comment by brophdawg11** [comment]
👍 I'm almost positive this is just a mismatched dependency/build issue which is why it only happened when upgrading from `1.11.0 -> 1.12.0` but is not reproducible on a fresh `1.12.0` install.  With the enhanced logging being released in `1.14.0` the underlying cause of these build issues should be better surfaced when running `remix dev`.

**Comment by pcattori** [comment]
Closing as this should have been addressed by #5441 .
